pkg/agent/tcpconn/disruptor.go

```diff
@@ -0,0 +1,84 @@
+package tcpconn
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/grafana/xk6-disruptor/pkg/iptables"
+	"github.com/grafana/xk6-disruptor/pkg/runtime"
+)
+
+// Disruptor applies TCP Connection disruptions.
+type Disruptor struct {
+	Executor   runtime.Executor
+	Handler    Handler
+	Disruption Disruption
+	NFQConfig  NFQConfig
+}
+
+type Disruption struct {
+	// Port is the target port to match which connections will be intercepted.
+	Port uint
+	// DropRate is the rate in [0, 1] range of connections that should be dropped.
+	DropRate float64
+	// DropPeriod defines how often connections will be terminated according to DropRate.
+	// If set to zero, connections will only be terminated once. If it is set to a non-zero value, connections will be
+	// re-evaluated for termination when this period passes.
+	DropPeriod time.Duration
+}
+
+// Apply executes the configured disruption for the specified duration.
+func (d Disruptor) Apply(ctx context.Context, duration time.Duration) error {
+	iptables := iptables.New(d.Executor)
+	defer iptables.Remove()
+
+	for _, r := range d.rules() {
+		err := iptables.AddRule(r)
+		if err != nil {
+			return err
+		}
+	}
+
+	// Set context timeout only if duration is not 0 (forever).
```

Comment:
I don't think we use this convention of duration `0` means forever in other disruptors. Worth checking we are consistent here

Reply:
Also, from the perspective of the JS API unlimited duration does not make sense, but for the agent as a stand-alone component it probably does, so maybe we can allow this value in all commands but limit it in the JS API. What do you think?